sympy/physics/vector/point.py

```diff
@@ -564,15 +564,7 @@ def vel(self, frame):
                             continue
                         candidate_neighbor.append(neighbor)
                         if not valid_neighbor_found:
-                            vel = None
-                            for f in self.pos_from(neighbor).args:
-                                if f[1] in self._vel_dict.keys():
-                                    if self._vel_dict[f[1]] != 0:
-                                        vel = self._vel_dict[f[1]]
-                                        break
-                            if vel is None:
-                                vel = self.pos_from(neighbor).dt(frame)
-                            self.set_vel(frame, vel + neighbor_velocity)
```

Comment:
I think instead of removing this code, we should write a more robust code that handles all cases.

Removing this code might break joints test.

Will do more detailed review, once I'm home.(will take 2-3 days)

Reply:
> Removing this code might break joints test.

@TJStienstra showed that (at least one) of the joints tests are incorrect.

Reply:
Besides the failing test. The code seems to me like a shortcut for computing the velocity. Which means that after removing it, it should in fact still work. Unless taking the derivative of the position vector, leads to a wrong answer, which can only be caused by:
- Mistake in the derivative function (unlikely)
- Mistake in the definition of someones position vector

Both these problems don't seem to be the case and otherwise would form a separate issue.
> I think instead of removing this code, we should write a more robust code that handles all cases.
With having a shortcut I do agree, but the only option I see for that is checking if we can use the `v1pt_theory`:
````py
vel = None
for interframe, frame_vel in neighbor._vel_dict.items():
    # Other option is to calculate self.vel(interframe), which gives recursions
    if frame_vel==0 and interframe in self._vel_dict:
        vel = self._vel_dict[interframe] + (interframe.ang_vel_in(frame) ^ self.pos_from(neighbor))
if vel is None:
    vel = self.pos_from(neighbor).dt(frame)
self.set_vel(frame, vel + neighbor_velocity)
```
Not sure if this is in the end really better. Also is there a reason why almost no recursive structures are used in these kind of calculations (just something I noticed)?